### PR TITLE
CBQI BV: Add inverse for more operators.

### PR DESCRIPTION
**Pull request by aniemetz**
This implements side conditions and the instantiation via word-level inversion for operators BITVECTOR_AND, BITVECTOR_OR, BITVECTOR_UREM (Index=1), BITVECTOR_LSHR (index=0).

### CODE REVIEW

**Comment by ajreynol** [comment]
Isn't BITVECTOR_ULT BV x BV -> Bool?  bv_solve_constraint involves only operators that are BV x BV -> BV.  Are you sure that your side condition code for BITVECTOR_ULT is getting called?

**Comment by ajreynol** [comment]
Btw, bit-vector predicates are currently handled in bv_solve_lit.  We could merge these functions if you want, but I figured they should remain separate.

File: src/theory/quantifiers/bv_inverter.cpp [hunk 1]
```diff
@@ -18,11 +18,30 @@
...
+static Node dropChild (Node n, unsigned index)
+{
+  unsigned nchildren = n.getNumChildren();
+  Assert (index < nchildren);
+  Kind k = n.getKind();
+  Assert (k == AND || k == OR || k == BITVECTOR_MULT || k == BITVECTOR_PLUS);
+  NodeBuilder<> nb (NodeManager::currentNM(), k);
+  for (size_t i = 0, nchildren = n.getNumChildren(); i < nchildren; ++i)
+  {
+    if (i == index) continue;
```
Comment (4tXJ7f) [inline src/theory/quantifiers/bv_inverter.cpp]: Why is `i` of type `size_t` and `index` of type `unsigned`?

**Comment by aniemetz** [comment]
It was indeed not called, you're right. I removed the code for ULT for now.

File: src/theory/quantifiers/bv_inverter.cpp (hunk 1 above)
Comment (aniemetz) [reply src/theory/quantifiers/bv_inverter.cpp]: No particular reason. Will change size_t to unsigned.

File: src/theory/quantifiers/bv_inverter.cpp [hunk 2]
```diff
@@ -18,11 +18,30 @@
...
+/* Drop child at given index from expression.
+ * E.g., dropChild((x + y + z), 1) -> (x + z)  */
+static Node dropChild (Node n, unsigned index)
+{
+  unsigned nchildren = n.getNumChildren();
+  Assert (index < nchildren);
+  Kind k = n.getKind();
+  Assert (k == AND || k == OR || k == BITVECTOR_MULT || k == BITVECTOR_PLUS);
+  NodeBuilder<> nb (NodeManager::currentNM(), k);
+  for (unsigned i = 0, nchildren = n.getNumChildren(); i < nchildren; ++i)
```
Comment (mpreiner) [inline src/theory/quantifiers/bv_inverter.cpp]: nchildren is already set to n.getNumChildren(), no need to do it again.

Comment (aniemetz) [reply src/theory/quantifiers/bv_inverter.cpp]: You're right, that's a left over. I'll fix that.

**Pull request merged**